app/src/main/java/com/example/filemanager/AllFilesViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.filemanager
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Build
+import android.os.Environment
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import java.io.File
+
+class AllFilesViewModel(
+    @SuppressLint("StaticFieldLeak")
+    private val context: Context
+) : ViewModel() {
+
+    val files = MutableLiveData<List<ListModel>>()
+
+    fun fetchFiles() {
+        val path = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            System.getenv("EXTERNAL_STORAGE")
```

Comment:
It doesnt look like an API provided by the Android SDK to load files. Can do some more investigations. There must be a better API to get the path to external storage directory.